src/set.rs

```diff
@@ -107,5 +146,385 @@ where
     }
 }
 
-// TODO: Tests
-// TODO: Iterators
+// TODO: impl Debug for ConSet<T, S>
+
+impl<T, S> ConSet<T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+{
+    /// Returns an iterator through the elements of the set.
+    pub fn iter(&self) -> Iter<T, S> {
+        Iter {
+            inner: raw::iterator::Iter::new(&self.raw),
+        }
+    }
+}
+
+/// The iterator of the [`ConSet`].
+///
+/// See the [`iter`][ConSet::iter] method for details.
+pub struct Iter<'a, T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+{
+    inner: raw::iterator::Iter<'a, TrivialConfig<T>, S>,
+}
+
+impl<'a, T, S> Iterator for Iter<'a, T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+{
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        match self.inner.next() {
+            Some(t) => Some(t.clone()),
+            None => None,
+        }
+    }
+}
+
+impl<'a, T, S> IntoIterator for &'a ConSet<T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+{
+    type Item = T;
+    type IntoIter = Iter<'a, T, S>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+impl<'a, T, S> Extend<T> for &'a ConSet<T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+    S: BuildHasher,
+{
+    fn extend<I>(&mut self, iter: I)
+    where
+        I: IntoIterator<Item = T>,
+    {
+        for n in iter {
+            self.insert(n);
+        }
+    }
+}
+
+impl<T, S> Extend<T> for ConSet<T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+    S: BuildHasher,
+{
+    fn extend<I>(&mut self, iter: I)
+    where
+        I: IntoIterator<Item = T>,
+    {
+        let mut se: &ConSet<_, _> = self;
```

Comment:
nitpick: What does `se` stand for?

In the map, `me` doesn't stand for anything, it's just me as the adjective (because self is keyword and can't be shadowed like this).

Reply:
Oh, I see, of course; it makes sense. I'll change it for consistency.